frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -2715,6 +2718,23 @@ bool TypeInference::hasVarbitsOrUnions(const IR::Type* type) const {
     return false;
 }
 
+bool TypeInference::onlyBitsOrBitStructs(const IR::Type* type) const {
+    // called for a canonical type
+    if (type->is<IR::Type_Bits>()) {
+        return true;
+    } else if (auto ht = type->to<IR::Type_Struct>()) {
+        for (auto f : ht->fields) {
+            auto ftype = typeMap->getType(f);
+            if (ftype == nullptr)
+                BUG("onlyBitsOrBitStructs check could not find type for %1%", f);
```

Comment:
You can use BUG_CHECK instead of if followed by BUG.

Reply:
will do.

Reply:
Ok, I will remove the flattenInterfaceStructs pass from p4test's midend.cpp and re-run 'make check'

Reply:
I have checked in new p4test/midend.cpp and also typeChecker.cpp to address the comments from today.  Please review.

Travis has passed.